pkg/controller/pxc/users.go

```diff
@@ -201,7 +202,7 @@ func (r *ReconcilePerconaXtraDBCluster) handleRootUser(ctx context.Context, cr *
 		Hosts: []string{"localhost", "%"},
 	}
 
-	if cr.Status.Status != api.AppStateReady {
+	if cr.Status.Status != api.AppStateReady && !r.invalidPasswordApplied(cr.Status) {
```

Comment:
how can user recover from this? is the message in status gone once the user updates the secret with a valid password?

Reply:
Normally if the app state is not ready we don't continue with the user update, but now we will not continue if the app state is not ready AND if the error is not invalid user.
Meaning, that if the state is not ready and the error is invalid user we will continue to update the user. We will try it until it is all good. In this case, until the password is valid.